tests/shared/src/test/scala/cats/mtl/tests/HandleTests.scala

```diff
@@ -39,4 +46,80 @@ class HandleTests extends BaseSuite {
     Handle[Kleisli[Foo, Unit, *], String]
     Handle[WriterT[Foo, String, *], String]
   }
+
+  test("submerge custom errors") {
+    sealed trait Error extends Product with Serializable
+
+    object Error {
+      case object First extends Error
+      case object Second extends Error
+      case object Third extends Error
+    }
+
+    val test =
+      Handle.ensure[F, Error](implicit h => Error.Second.raise.as("nope")) recover {
+        case Error.First => "0".pure[F]
+        case Error.Second => "1".pure[F]
+        case Error.Third => "2".pure[F]
+      }
+
+    assert(test.value.value.toOption == Some("1"))
+  }
+
+  test("submerge two independent errors") {
+    sealed trait Error1 extends Product with Serializable
+
+    object Error1 {
+      case object First extends Error1
+      case object Second extends Error1
+      case object Third extends Error1
+    }
+
+    sealed trait Error2 extends Product with Serializable
+
+    val test = Handle.ensure[F, Error1] { implicit h1 =>
+      Handle.ensure[F, Error2] { implicit h2 =>
+        val _ =
+          h2 // it's helpful to test the raise syntax infers even when multiple handles are present
+        Error1.Third.raise.as("nope")
+      } recover { e => e.toString.pure[F] }
+    } recover {
+      case Error1.First => "first1".pure[F]
+      case Error1.Second => "second1".pure[F]
+      case Error1.Third => "third1".pure[F]
+    }
+
+    assert(test.value.value.toOption == Some("third1"))
+  }
+
+  {
```

Comment:
Can it be? I have no idea how that works in MUnit.

Reply:
I'm confused :) is this a test or not haha, that is really what I am asking 😂

Reply:
It's a lexical block which contains Discipline stuff, which is many tests.

Reply:
Oh I see now, sorry. I was confused by the end of the block, it looked like an ordinary test ... but it's because the discipline tests have to exist _inside_ of the `Eval` since that's how the `Handle` is constructed. Thanks!

https://github.com/typelevel/cats-mtl/blob/0b8e8975d4c4722550c9bd9305df5a1ab0c0fd2b/tests/shared/src/test/scala/cats/mtl/tests/HandleTests.scala#L122-L124